Packages/com.unity.inputsystem/InputSystem/Controls/QuaternionControl.cs

```diff
@@ -66,17 +67,61 @@ protected override void FinishSetup()
         /// <inheritdoc/>
         public override unsafe Quaternion ReadUnprocessedValueFromState(void* statePtr)
         {
-            return new Quaternion(x.ReadValueFromState(statePtr), y.ReadValueFromState(statePtr), z.ReadValueFromState(statePtr),
-                w.ReadUnprocessedValueFromState(statePtr));
+            EnsureOptimizationTypeHasNotChanged();
+
+            switch (m_OptimizedControlDataType)
+            {
+                case InputStateBlock.kFormatQuaternion:
+                    return *(Quaternion*)((byte*)statePtr + (int)m_StateBlock.byteOffset);
+                default:
+                    return new Quaternion(
+                        x.ReadValueFromState(statePtr),
+                        y.ReadValueFromState(statePtr),
+                        z.ReadValueFromState(statePtr),
+                        w.ReadUnprocessedValueFromState(statePtr));
```

Comment:
If there was a conclusion why we have this odd patern of w not being processed, maybe add a comment explaining/motivating it if information was fresh?

Reply:
We have no idea why is that, and we have no idea if anyone is relying on this. So I can't add a comment